source/dem/pw_contact_force.cc

```diff
@@ -97,6 +97,57 @@ PWContactForce<dim>::update_contact_information(
   contact_info.tangential_overlap           = modified_tangential_overlap;
   contact_info.tangential_relative_velocity = tangential_relative_velocity;
 }
+template <int dim>
+void
+PWContactForce<dim>::calculate_force_torque_on_boundary_function(
+  const unsigned int boundary_id,
+  Tensor<1, dim>     add_force,
+  const Point<dim>   point_contact)
+{
+  if (calculate_force_torque_on_boundary == true)
+    {
+      force_on_walls[boundary_id] = force_on_walls[boundary_id] - add_force;
+      if (dim == 2)
+        {
+          Tensor<1, dim> r = point_contact - center_mass_container;
+          torque_on_walls[boundary_id][2] =
+            torque_on_walls[boundary_id][2] -
```

Comment:
I think torque is meaningless in two-dimensional simulations. What do you think @blaisb ?

Reply:
Well torque in 2D simulation is just the torque acting on the third axis, so it had a meaning even if it's a bit of a trivial meaning.